Resolves issue 678 - Clean up else condition to catch ALL remaining possibilities

fedebuyito opened the pull request:
## Description

Fix #678: The workflow was not catching condition when QR code (on seed exporting for backup confirmation step) is recognized but is not a SeedQR. Cleaning up else condition enables to catch ALL possibilities. Suggested by @kdmukai and Discuted with @newtonick on: https://github.com/SeedSigner/seedsigner/pull/677#discussion_r1938391630

Updated seed_flow test adding one of this cases (psbt), too.


_Include screenshots of any new or modified screens (or at least explain why they were omitted)_

This pull request is categorized as a:

- [ ] New feature
- [X] Bug fix
- [ ] Code refactor
- [ ] Documentation
- [ ] Other

## Checklist

- [X] I’ve run `pytest` and made sure all unit tests pass before sumbitting the PR

If you modified or added functionality/workflow, did you add new unit tests?

- [ ] No, I’m a fool
- [X] Yes
- [ ] N/A

I have tested this PR on the following platforms/os:

- [ ] Raspberry Pi OS [Manual Build](https://github.com/SeedSigner/seedsigner/blob/dev/docs/manual_installation.md)
- [ ] [SeedSigner OS](https://github.com/SeedSigner/seedsigner-os) on a Pi0/Pi0W board
- [X] Other


Note: Keep your changes limited in scope; if you uncover other issues or improvements along the way, ideally submit those as a separate PR. The more complicated the PR the harder to review, test, and merge.

fedebuyito committed:
Update seed_views.py

Cleanup the else condition so that the return to SeedTranscribeSeedQRConfirmInvalidQRView occurs on all remaining possibilities (any QR that is not SeedQR).

fedebuyito committed:
Update test_flows_seed.py

Includes additional failure condition (when QR is recognized but is not a SeedQR) in flow test, directed toward SeedTranscribeSeedQRConfirmInvalidQRView

alvroble commented:
Thanks @fedebuyito!!

Tested on Raspberry Pi OS.

The SeedTranscribeSeedQRConfirmInvalidQRView is displayed in all non-matching cases, including QR-encoded URLs, PSBT QRs, or random xpub QRs.

However, I noticed that SeedTranscribeSeedQRConfirmInvalidQRView also appears when pressing "Back" from SeedTranscribeSeedQRConfirmScanView. This issue persists in the latest dev commit. Pressing "Back" should navigate the user to the BackStackView (specifically, SeedTranscribeSeedQRConfirmQRPromptView). Since this is a minor bug in the dev branch, I suggest including this bugfix in the PR.

fedebuyito committed:
Merge branch 'SeedSigner:dev' into resolves_issue_678

fedebuyito committed:
Update scan_screens.py

Added a return value in the _run() method to evaluate whether the back keys are pressed when a QR code is scanned.

fedebuyito committed:
Update seed_views.py

Added a return to the last view in case the back button has been pressed in the SeedTranscribeSeedQRConfirmScanView class.

fedebuyito committed:
Update seed_views.py

Typo corrected

fedebuyito commented:
Hi @alvroble! Thank you for the review. 

Yes, you are right—good catch! I just updated this PR branch too. It seems to be a pre-existing bug, but I agree with solving it here. ACK regarding how the back button in that scan view should come back to SeedTranscribeSeedQRConfirmQRPromptView if scanning has not been executed.

Fixes have been made in the new commit and tested (Maybe you can check it, too):

![image](https://github.com/user-attachments/assets/a26bcf64-5666-47e1-a0d0-f85064f99444)


Regards!

alvroble commented:
Thanks for the update!! Now it's working right. ACK and tested on Raspberry Pi OS.

For the record, after seeing your code fix, I checked that something similar happens to all the classes that derive from `ScanView` in the `scan_views.py` file. They take you to the `MainMenuView` instead of the actual `BackStackView`. Although in those cases I'm not sure this is an actual bug or maybe a UX decision.

Regards!

newtonick commented:
tACK
